### PR TITLE
#161: Proposed fix

### PR DESCRIPTION
**Pull request by simon-brooke**
As discussed, this fix requires no change to existing themes. I shall be submitting a separate pull request to `cryogen` with rework to themes to generate better [OpenGraph](https://ogp.me/) meta-tags, but these changes do not depend on that rework.

### CODE REVIEW

**Review by yogthos — APPROVED** [review]

**Comment by yogthos** [comment]
Looks great overall, would you mind also updating the docs in the README.MD? :) Otherwise, I'm good to merge.

File: src/cryogen_core/util.clj [hunk 1]
```diff
@@ -60,3 +62,24 @@
...
+        c (count date-fmt)]
+    (try (if-let [last-slash (last-index-of file-name "/")]
+      (.parse fmt (.substring file-name (inc last-slash) (+ last-slash (inc c))))
+      (.parse fmt (.substring file-name 0 c)))
+         (catch Exception e
+           (throw (ex-info "Failed to parse date from filename"
+                          {:date-fmt date-fmt
+                           :file-name file-name}
+                          e)))))
+)
```
Comment (yogthos) [inline src/cryogen_core/util.clj:80]: minor formatting nitpick, can we move the paren to the previous line? :)

Comment (simon-brooke) [reply src/cryogen_core/util.clj:80]: Will do.

**Comment by simon-brooke** [comment]
OK, not clear which README you want changes in. I've pushed an update to `cryogen-project.cryogen` adding config documentation to the README; there's no documentation in the README of `cryogen-project.cryogen-core`. I do need to document the changes to allow meta-data to be inferred from plain Markdown (and presumably asciidoc, too, although I've not checked this), but that needs to be a change to [this page](http://cryogenweb.org/docs/writing-posts.html) and I'm not sure where to edit that!

**Comment by yogthos** [comment]
Ah yeah it would probably be best to update the docs site since I assume that's what most people will look at, the source for the site is here https://github.com/cryogen-project/cryogen-docs/tree/master/content/md/docs

**Comment by simon-brooke** [comment]
Sorry! Found that and sent you a pull request on that one now!

While I have your attention, where in the source code does the markdown parser get instantiated? I've written a function to detect whether there are any klipse code snippets in the document, and I'm trying to write unit tests for it, but I cannot find how to parse markdown text!

I see you wrote a markdown parser, but that's not a dependency; and I can't find any other dependency which implements markdown (or asciidoc) parsing.

**Comment by yogthos** [comment]
ok we should be all set, just pushed out new core and template to Clojars, thanks again for the pr :)

**Pull request merged**